gamemodes/terrortown/gamemode/client/cl_main.lua

```diff
@@ -867,12 +869,25 @@ function GM:DynamicCamera(viewTable, ply)
 
     -- dynamic mode: transition between different FOV values
     else
-        local time = math.max(
-            0,
-            (CurTime() - ply:GetFOVTime()) * game.GetTimeScale() * cvHostTimescale:GetFloat()
-        )
+        local fovTime = ply:GetFOVTime()
+
+        if lastFovTimeValue ~= fovTime then
+            lastFovTimeValue = fovTime
+            curTimeShift = 0
+        end
+
+        local curTime, fovTranDur = CurTime() + curTimeShift, ply:GetFOVTransitionTime()
```

Comment:
```suggestion
        local curTime = CurTime() + curTimeShift
        local fovTransitionDuration = ply:GetFOVTransitionTime()
```

Reply:
Done
I also realised I didn't need fovTransitionDuration in a local anymore